src/main/java/org/cactoos/io/Zip.java

```diff
@@ -33,50 +33,54 @@
 import java.util.zip.ZipEntry;
 import java.util.zip.ZipOutputStream;
 import org.cactoos.Input;
+import org.cactoos.func.ForEach;
 
 /**
  * Zip files and directory.
  * <br>
  * <br>There is no thread-safety guarantee.
  *
  * @since 0.29
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ * @todo #898:30min Currently {@link Zip} does not support zipping of empty
+ *  sub folder. Please change the implementation so that empty folders are
+ *  zipped as well.
  */
 @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
 public final class Zip implements Input {
 
     /**
-     * Origin directory.
+     * Origin list of files.
      */
-    private final Directory origin;
+    private final Directory directory;
 
     /**
      * Ctor.
      *
-     * @param origin Origin directory.
+     * @param dir Origin directory.
      */
-    public Zip(final Directory origin) {
-        this.origin = origin;
+    public Zip(final Directory dir) {
+        this.directory = dir;
     }
 
     @Override
     public InputStream stream() throws Exception {
         final ByteArrayOutputStream out = new ByteArrayOutputStream();
         try (ZipOutputStream zip = new ZipOutputStream(out)) {
-            for (final Path path : this.origin) {
-                final File file = path.toFile();
-                final ZipEntry entry = new ZipEntry(
-                    file.getPath()
-                );
-                zip.putNextEntry(entry);
-                if (file.isFile()) {
```

Comment:
@fanifieiev actually, now that I read this change in its greater context, I realise that the code before your changes was perfectly ok with the `closeEntry()` were it was because it did the `putNextEntry` entry BEFORE the `if`.

I think we should revert to the original behaviour and put back the test for empty directories, everything should be working without problem and there is no need for the `@todo`.

Reply:
@victornoel Ok, will revert 2 latest commits then

Reply:
@fanifieiev that's not what I said, I'm sorry I wasn't clear:
- you don't need the else branch
- you just need to put back the `putNextEntry` before the `if`
- and keep the `closeEntry` out of the `if` in that case.

In other words: you should revert back to the original code before any of your changes.

And from there:
- only add the relativisation of the path as required by the issue you are solving.
- if you want you can add the foreach for the loop, it makes sense
- then you can use the new tests you made and leave the new one for empty directories.
- if the test for empty directories does not pass, then you can add a todo

But please, do not try to tackle more, looking back, I think I should have refused the PR instead of taking so much of our time :( Sorry about that.

Reply:
@victornoel if I remove the else branch, the test for empty dir won't pass.
Actually, please refuse this PR, I will create another one later.
Thanks.